main.rb

```diff
@@ -0,0 +1,94 @@
+module Enumerable 
+	def my_each
+		for i in 0...(self.length)
+			yield(self[i])
+		end
+	end	
+
+	def my_each_with_index
+		for i in 0...self.length
+		   yield(self[i], i)
+		end
+	end
+
+	def my_select
+		arr = []
+		self.my_each {|item| arr.push(item) if yield(item) }
+		return arr
+	end
+
+	def my_all?
+		output = true
+		self.my_each {|item| output = false unless yield(item)}
+		return output
+	end
+
+	def my_any? 
+		output = false
+		self.my_each {|item| output = true if yield(item)}
+		return output
+	end
+
+	def my_none?
+		output = true
+		self.my_each { | item | output = false if yield(item) }
+		return output
+	end
+
+	def my_count(item = nil)
+      count = 0
+      if item != nil
+        self.my_each{ |element| count += 1 if element == item }
+      elsif block_given?
+        self.my_each{ |element| count += 1 if yield(element)}
+      else
+        count = self.length 
+      end
+      return count
+    end
+
+  def my_map(proc=nil)
+    	arr = []
+    	if proc == nil
+    	   self.my_each_with_index{ | element, index | arr << yield(element)}
+    	else
+    		self.my_each_with_index { |element, index| arr << proc.call(element)}
+    	end
+    	return arr
+    end
+
+    def my_inject(initial = nil)
+      
+      if initial == nil 
+        accumulator = self.first
+      else
+        accumulator = initial
+      end
+
+      self.my_each {|element| accumulator = yield(accumulator,element)} 
```

Comment:
Wow, disappointed 😢 

You did not address my review comment on my_inject.

This method fails because of this line. I wonder why? 🤔 

Maybe when an initial value is not given you actually are supposed to accumulate from the second element in the array since initial accumulator includes the first element already. So this line would only be true if initial is not nil.

so something like `self[1..-1]` could be used.

Does this imply two statements of this kind needed maybe?

if initial is nil then your accumulator starts with self.first which means you should only run the my_each block from the second element in the array.

Otherwise, if there is an initial then you run on the whole array.... does this make sense?